minidump/src/minidump.rs

```diff
@@ -3946,8 +3946,21 @@ fn read_string_list(
     let count: u32 = data
         .gread_with(&mut offset, endian)
         .or(Err(Error::StreamReadFailure))?;
+    let count = count as usize;
+    // NOTE: this only counts the RVA size, not the raw variable-size string data,
+    // but that should be good enough for the bounds check
+    let counted_size = match count.checked_mul(<md::RVA>::size_with(&endian)) {
+        Some(s) => s,
+        None => return Err(Error::StreamReadFailure),
```

Comment:
You might alternately write this as:

```rust
let counted_size = count.checked_mul(<md::RVA>::size_with(&endian)).ok_or(Error::StreamReadFailure)?;
```

Reply:
Oh, absolutely!

I will try making this a more generic util along with that improvement.